core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -163,6 +167,24 @@ abstract class AppBase(
     }
   }
 
+  def calculateTotalCoreMs(): Unit = {
+    executorIdToInfo.foreach { case(_, eInfo) =>
+      val eStartTime = eInfo.addTime.getTime
+      var eEndTime = eInfo.removeTime
+      if (eEndTime == 0L) {
+        getAppEndTime match {
+          case Some(appEndTime) =>
```

Comment:
at this point the appEndTime has been estimated, correct?  Meaning if file truncated we use last time (job end or sql end) time.

Reply:
+1
the `appEndTime` has been estimated inside `calculateAppDuration()`. There is a corner case where the appEndTime would still be missing. If `appMetaData` is not defined, then there won't be an `appEndTime`.
I can't think of a situation where we have `executorInfo` defined without having `appMetaData`..

Reply:
yes, `appEndTime` is estimated before calculating total core seconds. For the corner case @amahussein mentioned, do we want to follow up with that?

Reply:
@amahussein It is unlikely and let's say in case of an incomplete event log that happens, we will eventually throw `IncorrectAppStatusException` exception if app metadata is undefined.

Reply:
> do we want to follow up with that @amahussein ?
No no need to.